libraries/AP_RangeFinder/AP_RangeFinder.cpp

```diff
@@ -71,6 +71,8 @@
 #include <AP_Vehicle/AP_Vehicle_Type.h>
 #include <AP_HAL/I2CDevice.h>
 #include <AP_InternalError/AP_InternalError.h>
+#include <AP_TemperatureSensor/AP_TemperatureSensor.h>
```

Comment:
This should be guarded with a precompiler check on whether the temperature sensor library is enabled - otherwise it shouldn’t be included in the build.

Reply:
This is why we have the _config.h files. It is a lightweight way to check/populate the define to see if a library is enabled, then you can check the define to include the main .h